src/hooks/useVenueAccess.ts

```diff
@@ -22,33 +22,37 @@ export const useVenueAccess = (
     onDenyAccess && onDenyAccess();
   }, [onDenyAccess, venue]);
 
+  const logout = useCallback(() => {
+    firebase
+      .auth()
+      .signOut()
+      .finally(() => {
+        denyAccess();
+      });
+  }, [denyAccess]);
+
   useEffect(() => {
     if (!venue) return;
 
     if (venue.access) {
-      const token = getLocalStorageToken(venue.id) ?? undefined;
-      if (!token) {
+      const token = getLocalStorageToken(venue.id);
+      if (token) {
         checkAccess({
           venueId: venue.id,
           token,
         }).then((result) => {
           if (!isTruthy(result.data)) {
-            firebase
-              .auth()
-              .signOut()
-              .finally(() => {
-                denyAccess();
-              });
+            logout();
           }
         });
       } else {
         firebase
           .auth()
           .signOut()
           .finally(() => {
-            denyAccess();
+            logout();
           });
```

Comment:
I think this whole chunk should just be `logout`? Currently it signs out, to then sign out, to then deny access.